## Producción con proceso I, II y III, Demanda máxima de C es 10 ton/hr
___
**Código de colores:**
<font color = red>Proceso I</font>   
<font color = green>Proceso II</font>  
<font color = blue>Proceso III</font>  
<font color = yellow>Demanda de C hasta 10 ton/hr</font>  
<font color = Purple>Demanda de C hasta 15 ton/hr</font>  

**Ganancía Total:**  
$$-(-(950B_0) \color{red}{-(500A_1)-(1000N)-(250A_1)} \color{green}{-(1500N)-(400B_2)}\color{blue}{-(2000N)-(550B_3)}\color{yellow}{+(1800CD1)})$$  

**Restricciones:**
$$\color{red}{B_1-0.9A_1 = 0}$$
$$B_2+B_3-B_1-B_0 = 0$$
$$\color{green}{C_2-0.82B_2 = 0}$$
$$\color{blue}{C_3-0.95B_3= 0}$$
$$\color{yellow}{CD_1-C_2-C_3 = 0}$$
$$\color{red}{A_1 \leq 16}$$
$$\color{yellow}{CD_1 \leq 10}$$
$$\color{yellow}{-CD_1 \leq 0}$$
$$N \leq 1$$
$$-N \leq -1$$

In [1]:
import scipy.optimize as opt
import numpy as np
C = np.array([750, 950, 0, 400, 550, 0, 0, -1800, 4500])
Ae = np.array([[-0.9,0,1,0,0,0,0,0,0],
               [0,-1,-1,1,1,0,0,0,0],
               [0,0,0,-0.82,0,1,0,0,0],
               [0,0,0,0,-0.95,0,1,0,0],
               [0,0,0,0,0,-1,-1,1,0]])
be = np.array([0,0,0,0,0])
Ad = np.array([[1,0,0,0,0,0,0,0,0],
               [0,0,0,0,0,0,0,1,0],
               [0,0,0,0,0,0,0,-1,0],
               [0,0,0,0,0,0,0,0,1],
               [0,0,0,0,0,0,0,0,-1]])
bd = np.array([16,10,0,1,-1])
resultado = opt.linprog(C, A_ub=Ad, b_ub=bd, A_eq=Ae, b_eq=be)
resultado


     fun: 1061.40350877193
 message: 'Optimization terminated successfully.'
     nit: 9
   slack: array([ 4.30409357,  0.        , 10.        ,  0.        ,  0.        ])
  status: 0
 success: True
       x: array([1.16959064e+01, 0.00000000e+00, 1.05263158e+01, 0.00000000e+00,
       1.05263158e+01, 1.77635684e-15, 1.00000000e+01, 1.00000000e+01,
       1.00000000e+00])

In [2]:
import pandas as pd
info = pd.DataFrame({'A1': [resultado.x[0]], 'B0': [resultado.x[1]], 'B1': [resultado.x[2]], 'B2': [resultado.x[3]], 'B3': [resultado.x[4]], 'C2': [resultado.x[5]], 'C3': [resultado.x[6]], 'CD1': [resultado.x[7]], 'Ganancia': [-resultado.fun]}, index = ['Variables'])
info

A1   B0         B1   B2         B3            C2    C3  \
Variables  11.695906  0.0  10.526316  0.0  10.526316  1.776357e-15  10.0   

            CD1     Ganancia  
Variables  10.0 -1061.403509

Los resultados de la tabla muestran que no se comprara nada de B del mercado y en cambio todo sera producido a partir del proceso I, a su vez la formación de C se realizara únicamente por el proceso III.  
Debido a que aunque no se usó el proceso II para la producción de C y aun así  se contempla su gasto por hora de uso, el resultado índica que se genera una perdida de 1061, este debe de ser calculado denuevo con un sistma que únicamente contemple los gastos generados por el proceso I y III.

## Producción con proceso I y III, Demanda máxima de C es 10 ton/hr
___
**Código de colores:**
<font color = red>Proceso I</font>   
<font color = green>Proceso II</font>  
<font color = blue>Proceso III</font>  
<font color = yellow>Demanda de C hasta 10 ton/hr</font>  
<font color = Purple>Demanda de C hasta 15 ton/hr</font>  

**Ganancía Total:**  
$$-(-(950B_0) \color{red}{-(500A_1)-(1000N)-(250A_1)}\color{blue}{-(2000N)-(550B_3)}\color{yellow}{+(1800CD1)})$$  

**Restricciones:**
$$\color{red}{B_1-0.9A_1 = 0}$$
$$B_3-B_1-B_0 = 0$$
$$\color{blue}{C_3-0.95B_3 = 0}$$
$$\color{yellow}{CD_1-C_3 = 0}$$
$$\color{red}{A_1 \leq 16}$$
$$\color{yellow}{CD_1 \leq 10}$$
$$\color{yellow}{-CD_1 \leq 0}$$
$$N \leq 1$$
$$-N \leq -1$$

In [1]:
import scipy.optimize as opt
import numpy as np
C = np.array([750, 950, 0, 550, 0, -1800, 3000])
Ae = np.array([[-0.9,0,1,0,0,0,0],
               [0,-1,-1,1,0,0,0],
               [0,0,0,-0.95,1,0,0],
               [0,0,0,0,-1,1,0]])
be = np.array([0,0,0,0])
Ad = np.array([[1,0,0,0,0,0,0],
               [0,0,0,0,0,1,0],
               [0,0,0,0,0,-1,0],
               [0,0,0,0,0,0,1],
               [0,0,0,0,0,0,-1]])
bd = np.array([16,10,0,1,-1])
resultado = opt.linprog(C, A_ub=Ad, b_ub=bd, A_eq=Ae, b_eq=be)
resultado

     fun: -438.59649122807014
 message: 'Optimization terminated successfully.'
     nit: 7
   slack: array([ 4.30409357,  0.        , 10.        ,  0.        ,  0.        ])
  status: 0
 success: True
       x: array([11.69590643,  0.        , 10.52631579, 10.52631579, 10.        ,
       10.        ,  1.        ])

In [2]:
import pandas as pd
info = pd.DataFrame({'A1': [resultado.x[0]], 'B0': [resultado.x[1]], 'B1': [resultado.x[2]], 'B3': [resultado.x[3]], 'C3': [resultado.x[4]], 'CD1': [resultado.x[5]], 'Ganancia': [-resultado.fun]}, index = ['Variables'])
info

A1   B0         B1         B3    C3   CD1    Ganancia
Variables  11.695906  0.0  10.526316  10.526316  10.0  10.0  438.596491

Los resultados de la tabla muestran que no se comprara nada de B del mercado y en cambio todo sera producido a partir del proceso I, a su vez la formación de C se realizara únicamente por el proceso III.  
Esto generaría una ganancia de 438.59/hr con una demanda de C de 10 ton/hr.
Sin embargo hace falta comprobar si al cobrar 1500/ton en las demandas de C de 10-15ton/hr pueden aumentar las ganancias de producción.